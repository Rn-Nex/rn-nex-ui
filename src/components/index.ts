export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
