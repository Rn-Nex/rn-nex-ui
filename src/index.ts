export * from './components';
export * from './libraries';
export * from './utils';
