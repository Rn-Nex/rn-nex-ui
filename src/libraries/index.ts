export * from './themes/v2';
export * from './style';
