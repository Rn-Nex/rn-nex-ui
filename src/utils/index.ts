export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
