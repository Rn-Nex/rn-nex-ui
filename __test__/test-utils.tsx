import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
