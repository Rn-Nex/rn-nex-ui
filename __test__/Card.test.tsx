import React from 'react';
import { View } from 'react-native';
import { Card, CardContent, CardHeader, grey } from '../src';
import { render } from './test-utils';

describe('Card Component', () => {
  const mockCardTestId = 'card-test-id';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<Card />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<Card ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should apply the sx styles correctly', () => {
    const mockSx = { bg: 'red' };
    const { getByTestId } = render(<Card sx={mockSx} testID={mockCardTestId} />);

    const card = getByTestId(mockCardTestId);
    expect(card.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });

  it('should apply the card variant (outlined)', () => {
    const { getByTestId } = render(<Card variation="outlined" testID={mockCardTestId} />);

    const card = getByTestId(mockCardTestId);
    expect(card.props.style).toEqual(expect.objectContaining({ borderWidth: 0.5, borderColor: grey[500] }));
  });
});

describe('CardHeader Component', () => {
  const mockCardHeaderTestId = 'card-header-test-id';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<CardHeader />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<CardHeader ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should apply the sx styles correctly', () => {
    const mockSx = { bg: 'red' };
    const { getByTestId } = render(<CardHeader sx={mockSx} testID={mockCardHeaderTestId} />);

    const cardHeader = getByTestId(mockCardHeaderTestId);
    expect(cardHeader.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });
});

describe('CardContent Component', () => {
  const mockCardContentTestId = 'card-content-test-td';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<CardContent />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<CardContent ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should apply the sx styles correctly', () => {
    const mockSx = { bg: 'red' };
    const { getByTestId } = render(<CardContent sx={mockSx} testID={mockCardContentTestId} />);

    const cardHeader = getByTestId(mockCardContentTestId);
    expect(cardHeader.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });
});
