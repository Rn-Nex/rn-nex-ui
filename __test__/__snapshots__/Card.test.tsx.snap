// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly 1`] = `[Function]`;

exports[`CardContent Component should render correctly 1`] = `[Function]`;

exports[`CardHeader Component should render correctly 1`] = `[Function]`;
